integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -410,21 +410,50 @@ def create_empty_df(spark, path):
         data_path,
         conf=writer_confs)
 
-# should fallback when trying to write field ID metadata
+def get_nested_parquet_meta_data_for_field_id():
+    schema = StructType([
+        StructField("c1", IntegerType(), metadata={'parquet.field.id': -1}),
+        StructField("c2", StructType(
+            [StructField("c3", IntegerType(), metadata={'parquet.field.id': -3})]),
+                    metadata={'parquet.field.id': -2})
+    ])
+    data = [(1, (2,)), (11, (22,)), (33, (33,)), ]
+    return schema, data
+
+
 @pytest.mark.skipif(is_before_spark_330(), reason='Field ID is not supported before Spark 330')
-@allow_non_gpu('DataWritingCommandExec')
 def test_parquet_write_field_id(spark_tmp_path):
+    data_path = "/tmp/m1"
```

Comment:
Should use `spark_tmp_path` instead of a hardcoded path here.

Reply:
Done